client/client.go

```diff
@@ -116,6 +116,17 @@ type SysInfo struct {
 	Store            string `json:"store,omitempty"`
 }
 
+// processErrorResponse handles the common error path for API requests.
+// This function should be called when response.Type == "error"
+func (rsp *response) processErrorResponse() error {
```

Comment:
Because this leads to "cannot do stuff: cannot do stuff", the error already has the useful message.